lib/truffle/truffle/patching.rb

```diff
@@ -0,0 +1,38 @@
+module Truffle::Patching
```

Comment:
This approach is quite different than what I had in mind in https://github.com/graalvm/truffleruby/pull/68#issuecomment-280852060. Your approach could also use that mechanism, but it is simpler and sufficient for now, so let's keep it simple :+1: 
Could you add a `lib/patches/README.md` document briefly what it is for and what it does?